### PR TITLE
Release 1.12.0

### PR DESCRIPTION
**Pull request by bharanidharan14**
Update setup.cfg, docs/conf.py and CHANGELOG.rst with the changes pertaining to release 1.12.0

### CODE REVIEW

**Comment by bharanidharan14** [comment]
@phanikumv @rajaths010494 Can you review this PR

File: CHANGELOG.rst [hunk 1]
```diff
@@ -1,6 +1,32 @@
...
+Feature
+"""""""
+
+- Add custom XCom backend for GCS.
+  (`#681 <https://github.com/astronomer/astronomer-providers/pull/681>`_)
+
+Bug Fixes
+"""""""""
+
+- Change return value of ``SnowflakeOperatorAsync`` to be same as the Sync operator.
```
Comment (phanikumv) [inline CHANGELOG.rst:16]:
```suggestion
- Change return value of ``SnowflakeOperatorAsync`` to be same as that of ``SnowflakeOperator``
```

File: CHANGELOG.rst [hunk 2]
```diff
@@ -1,6 +1,32 @@
...
+  (`#781 <https://github.com/astronomer/astronomer-providers/pull/781>`_)
+- Add poke_interval to ``S3KeySensorAsync``.
+  (`#782 <https://github.com/astronomer/astronomer-providers/pull/782>`_)
+- Fix ``AwsBaseHookAsync`` to get connection details using ``AwsConnectionWrapper``.
```
Comment (phanikumv) [inline CHANGELOG.rst:20]: this sentence need to be from a user point of view

File: CHANGELOG.rst [hunk 3]
```diff
@@ -1,6 +1,32 @@
...
+  (`#758 <https://github.com/astronomer/astronomer-providers/pull/758>`_)
+- Remove mypy ignore specification in ``WasbHookAsync``.
```
Comment (phanikumv) [inline CHANGELOG.rst:22]: not needed as this is minor tech. change

Comment (bharanidharan14) [reply CHANGELOG.rst:22]: Ok, removing it

File: CHANGELOG.rst (hunk 1 above)
Comment (bharanidharan14) [reply CHANGELOG.rst:16]: Fixed

File: CHANGELOG.rst (hunk 2 above)
Comment (bharanidharan14) [reply CHANGELOG.rst:20]: Use ``AwsConnectionWrapper`` to get connection details to create AWS async client.

**Review by rajaths010494 — APPROVED** [review]

**Comment by bharanidharan14** [comment]
@phanikumv Addressed your review comment

**Review by phanikumv — APPROVED** [review]

**Pull request merged**